bionic/core/flow_execution.py

```diff
@@ -146,12 +146,12 @@ def _process_active_entry(self, entry):
         if self._mark_entry_completed_if_possible(entry):
             return
 
-        # If this entry is persistable, we may be able to load it from the persistent
-        # cache, which would immediately get us to the CACHED level.
-        if entry.state.should_persist:
-            entry.state.attempt_to_access_persistent_cached_value()
+        # If this entry produces an artifact, we may be able to load it from the
+        # persistent cache, which would immediately get us to the CACHED level.
+        if entry.state.yields_artifact:
```

Comment:
What about `produces_artifact` to match comments?

Reply:
In `task_execution.py`, we talk about the difference in behavior between persistable and non-persistable output. To make things even more clearer, what about something along the lines of `produces_persistable_artifact` or `output_is_persistable` ?